src/databricks/labs/ucx/azure/access.py

```diff
@@ -91,6 +105,115 @@ def save_spn_permissions(self) -> str | None:
+
+            self._installation.save(cluster_policy, filename="policy-backup.json")
+
+            if cluster_policy.definition is not None:
+                policy_definition = self._update_cluster_policy_definition(
+                    cluster_policy.definition, storage_accounts, uber_principal, inventory_database
+                )
+            if cluster_policy.name is not None:
+                self._ws.cluster_policies.edit(policy_id, cluster_policy.name, definition=policy_definition)
+        except NotFound:
+            msg = f"cluster policy {policy_id} not found, please run UCX installation to create UCX cluster policy"
+            raise NotFound(msg) from None
+
+    def create_uber_principal(self, prompts: Prompts):
+        config = self._installation.load(WorkspaceConfig)
+        inventory_database = config.inventory_database
+        display_name = f"unity-catalog-migration-{inventory_database}-{self._ws.get_workspace_id()}"
+        uber_principal_name = prompts.question(
+            "Enter a name for the uber service principal to be created", default=display_name
+        )
+        policy_id = config.policy_id
+        if policy_id is None:
+            msg = "UCX cluster policy not found in config. Please run latest UCX installation to set cluster policy"
+            logger.error(msg)
+            raise ValueError(msg) from None
+        if config.uber_spn_id is not None:
+            logger.error("Uber service principal already created for this workspace.")
+            return
+        used_storage_accounts = self._get_storage_accounts()
+        if len(used_storage_accounts) == 0:
+            logger.warning(
+                "There are no external table present with azure storage account. "
+                "Please check if assessment job is run"
+            )
+            return
+        storage_account_info = []
+        for storage in self._azurerm.storage_accounts():
+            if storage.storage_account in used_storage_accounts:
+                storage_account_info.append(storage)
+        logger.info("Creating service principal")
+        uber_principal = self._azurerm.create_service_principal(uber_principal_name)
+        config = self._installation.load(WorkspaceConfig)
+        config.uber_spn_id = uber_principal.client.client_id
+        self._installation.save(config)
+        logger.info(f"Creating secret scope {inventory_database}.")
+        try:
+            self._ws.secrets.create_scope(inventory_database)
+        except ResourceAlreadyExists:
+            logger.warning(f"Secret scope {inventory_database} already exists, using the same")
+        self._ws.secrets.put_secret(inventory_database, "uber_principal_secret", string_value=uber_principal.secret)
+        logger.info(
+            f"Created service principal of client_id {config.uber_spn_id}. " f"Applying permission on storage accounts"
+        )
+        for storage in storage_account_info:
+            role_name = str(uuid.uuid4())
```

Comment:
isn't it "role_assignment_id" instead?...

Reply:
there are two id's: role_name which is a unique guide for each assignment. then there is roleassignment id which is fixed for the blob data reader role